### ISSUE TITLE
QBSolv should be a composite

### ISSUE THREAD

**Issue by spakin** [dwavesystems/qbsolv#149]
**Current Problem**

It strikes me as inconsistent that the `QBSolv` class is not defined as a composite sampler.  If I want to add auto-embedding support to a sampler, I pass the sampler to an [`AutoEmbeddingComposite`](https://docs.ocean.dwavesys.com/en/stable/docs_system/reference/composites.html#autoembeddingcomposite).  If I want to add reverse-annealing support to a sampler, I pass the sampler to a [`ReverseAdvanceComposite`](https://docs.ocean.dwavesys.com/en/stable/docs_system/reference/composites.html#reverseadvancecomposite).  If I want to add spin-reversal support to a sampler, I pass the sampler to a [`SpinReversalTransformComposite`](https://docs.ocean.dwavesys.com/en/stable/docs_dimod/reference/sampler_composites/composites.html#dimod.reference.composites.spin_transform.SpinReversalTransformComposite).  But if I want to add QBSolv support to sampler to break up a large problem, I first create a [`QBSolv`](https://docs.ocean.dwavesys.com/projects/qbsolv/en/latest/source/python.html#dwave_qbsolv.QBSolv) object with no child sampler.  Only when I sample from the `QBsolv` object do I pass [`QBSolv.sample`](https://docs.ocean.dwavesys.com/projects/qbsolv/en/latest/source/generated/dwave_qbsolv.QBSolv.sample.html#dwave_qbsolv.QBSolv.sample) the sampler I want to use.

**Proposed Solution**
I propose deriving `QBSolv` from [`dimod.Composite`](https://docs.ocean.dwavesys.com/projects/dimod/en/latest/reference/sampler_composites/api.html#dimod.Composite).  That way, users can treat `QBSolv` as any other composite sampler, specifying the child at creation time, not at sampling time.

**Comment by arcondello** [comment]
I agree. QBSolve pre-dates the notion of a composite, but conceptually you are absolutely correct.

For another approach, I would check out [dwave-hybrid](https://github.com/dwavesystems/dwave-hybrid)'s [qbsolv-like](https://github.com/dwavesystems/dwave-hybrid/blob/master/examples/qbsolv-like.py) which will give you more control.
